regolith/helpers/l_todohelper.py

```diff
@@ -117,31 +116,45 @@ def sout(self):
                 todo["due_date"] = date_parser.parse(todo["due_date"]).date()
             todo["days_to_due"] = (todo.get('due_date') - today).days
             todo["order"] = todo['importance'] + 1 / (1 + math.exp(abs(todo["days_to_due"])))
-
-        gather_todos = sorted(gather_todos, key=lambda k: (-k['order']))
-
+        gather_todos[:len_of_tasks] = sorted(gather_todos[:len_of_tasks], key=lambda k: (-k['order'], k.get('duration', 10000)))
+        gather_todos[len_of_tasks:] = sorted(gather_todos[len_of_tasks:], key=lambda k: (-k['order'], k.get('duration', 10000)))
         if rc.short_tasks:
             for todo in gather_todos[::-1]:
                 if todo.get('duration') is None or float(todo.get('duration')) > float(rc.short_tasks):
-                    gather_todos.remove(t)
-        num = 1
+                    gather_todos.remove(todo)
         print("-" * 50)
-        print("    action (days to due date|importance|expected duration(mins))")
-        for todo in gather_todos:
-            if todo.get('status') not in ["finished", "cancelled"]:
+        print("  (running_index) action (days to due date|importance|expected duration(mins))")
```

Comment:
since we are reusing code in different places, it makes sense to break it out into a function or sthg and call that function in each place.  Much easier to maintain, and ensures the same behavior in each place.